python/pyarrow/tests/test_fs.py

```diff
@@ -291,6 +291,37 @@ def subtree_s3fs(request, s3fs):
 }"""
 
 
+@pytest.fixture
+def azurefs(request, azure_server):
+    request.config.pyarrow.requires('azure')
+    from pyarrow.fs import AzureFileSystem
+
+    host, port = azure_server['connection']
+    azureite_authority = f"{host}:{port}"
+    azureite_scheme = "http"
+
+    container = 'pyarrow-filesystem/'
+
+    fs = AzureFileSystem(account_name='devstoreaccount1',
+                         account_key='Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuF'
+                         'q2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==',
```

Comment:
Where does this account name and key come from?

Reply:
These are the default  storage account and account name for `azurite`. https://learn.microsoft.com/en-us/azure/storage/common/storage-use-emulator#authorize-with-shared-key-credentials

Reply:
Done

Reply:
For S3 the equivalent of account_name and account_key actually come from `s3_server`. I'm going to make Azure consistent